plugins/captcha/powcaptcha/src/Extension/POWCaptcha.php

```diff
@@ -0,0 +1,256 @@
+     *
+     * @throws  \RuntimeException
+     * @since __DEPLOY_VERSION__
+     */
+    public function onInit($id = 'altcha_1')
+    {
+        $app = $this->getApplication();
+
+        if (!$app instanceof CMSWebApplicationInterface) {
+            return false;
+        }
+
+        // Load assets
+        $app->getDocument()->getWebAssetManager()->usePreset('altcha');
+
+        return true;
+    }
+
+    /**
+     * Gets the challenge HTML
+     *
+     * @param   string  $name   The name of the field. Not Used.
+     * @param   string  $id     The id of the field.
+     * @param   string  $class  The class of the field.
+     *
+     * @return  string  The HTML to be embedded in the form.
+     *
+     * @since __DEPLOY_VERSION__
+     */
+    public function onDisplay($name = null, $id = 'altcha_1', $class = '')
+    {
+        $htmlAttributes = [
+            'name'       => $name,
+            'id'         => $id,
+            'class'      => $class,
+            'hidefooter' => true,
+            'hidelogo'   => true,
+            'auto'       => $this->params->get('autosolve', 'onfocus'),
+            'strings'    => htmlentities(
+                json_encode(
+                    [
+                        'ariaLinkLabel' => Text::_('PLG_CAPTCHA_POWCAPTCHA_ARIALINKLABEL'),
+                        'error'         => Text::_('PLG_CAPTCHA_POWCAPTCHA_ERROR'),
+                        'expired'       => Text::_('PLG_CAPTCHA_POWCAPTCHA_EXPIRED'),
+                        'footer'        => Text::_('PLG_CAPTCHA_POWCAPTCHA_FOOTER'),
+                        'label'         => Text::_('PLG_CAPTCHA_POWCAPTCHA_LABEL'),
+                        'verified'      => Text::_('PLG_CAPTCHA_POWCAPTCHA_VERIFIED'),
+                        'verifying'     => Text::_('PLG_CAPTCHA_POWCAPTCHA_VERIFYING'),
+                        'waitAlert'     => Text::_('PLG_CAPTCHA_POWCAPTCHA_WAITALERT'),
+                    ]
+                ),
+                ENT_QUOTES,
+                'UTF-8'
+            ),
+            'challengeurl' => Route::_(
+                \sprintf(
+                    "index.php?option=com_ajax&plugin=powcaptcha&group=captcha&format=raw&%s=1",
+                    Session::getFormToken()
+                ),
+                false,
+                false,
+                true
+            ),
+        ];
+
+        return \sprintf(
+            '<altcha-widget %s></altcha-widget>',
+            ArrayHelper::toString($htmlAttributes)
+        );
+    }
+
+    /**
+     * Verify the users answer
+     *
+     * @param   string  $code  Answer provided by user. Not needed for the Recaptcha implementation
+     *
+     * @return  bool  True if the answer is correct, false otherwise
+     *
+     * @throws  \RuntimeException
+     * @since __DEPLOY_VERSION__
+     */
+    public function onCheckAnswer($code = null)
+    {
+        // Before we verify the actual solution, let's first our challenge key
```

Comment:
There is a verb missing here. Guessing but perhaps it should be

```suggestion
        // Before we verify the actual solution, let's first verify our challenge key
```